src/libs/type-utils/lodash-fp-helpers.ts

```diff
@@ -0,0 +1,138 @@
+import _ from 'lodash/fp'
+import { AnyFn, AnyPromiseFn, GenericFn, WrapFn } from 'src/libs/type-utils/general-types'
+
+/**
+ * (async Promise overload) Provides better type flow and ergonomics for the common case of wrapping a
+ * main function (typically a data call that returns a promise) with
+ * additional pass-thru wrapper functions (handlers) to handle things like
+ * errors or retry.  These handlers are expected to be HoF's
+ * (higher order functions) that return a wrapped function with the desired
+ * handling logic which still honors the same function signature of the fn
+ * it is given as it's single argument
+ * @param handlers
+ * @param mainFn
+ */
+export function withHandlers<P, A extends any[], F extends (...args: A) => Promise<P>>(
```

Comment:
Given that functions in Terra UI are generally declared with `export const someFn = (...) => { ... }` instead of `export function someFn (...) { ... }`, would it be preferable to define overloads using an interface instead of multiple `export function` statements?

Something like how the types for `h` are declared:
https://github.com/DataBiosphere/terra-ui/blob/c31123e6328f56789203ba3a8f31e60786e61446/types/react-hyperscript-helpers/index.d.ts#L15-L30

Reply:
done, with great results!